Toolbar wrapping fix & create trail dialog positioning

bmcdougald opened the pull request:
Two sets of fixes on this ticket.  The first was to prevent the toggle button section from unnaturally wrapping when reducing window size.  The second was to maintain position of the trail text area and button locations when entering "create trail" mode.

bwhiteman commented:
There are still problems with the spacing. There is unused space on the left side of the toolbar yet you can have the toolbar aligned properly before login then loose the buttons after login and the domain/trail names get truncated.
<img width="1440" alt="screen shot 2016-01-15 at 2 23 48 pm" src="https://cloud.githubusercontent.com/assets/1516655/12362873/a65f2d14-bb93-11e5-8fbd-35bda1302a0b.png">
In this image, the window is just below fully maximized and the logout button isn't visible.

bmcdougald commented:
Hmm.  That's very odd.  I don't have that button issue on my vm or local box when hitting that page.  Can you try rebuilding and installing the Firefox plugin again?   I will look at this again Monday and we can chat about it.  As for the space on the left, we have plans for adding buttons to get directly to forensic and things to come that will go there like a real toolbar.

bmcdougald commented:
I was able to repro what Brandon showed in the screenshot.  There was also some strange things happening with the Domain and Trail labels when resizing.  I've adjusted styling to reactively resize down to a certain point using a min-width for the entire form along with the column sizes.  The buttons and controls no longer wrap to a second line where they cannot be used, instead they simply flow off the right margin of the toolbar when you've sized the window beyond a minimum size value.  This seemed the cleanest way of dealing with stuff not fitting short of putting scrollbars in the toolbar which would cause other visibility issues when it popped up.

I'd like to visit better styling overall in the toolbar at a later date, but wanted this fixed at a minimum before SUG testing.

bwhiteman commented:
This is better visually but still causes the problem where if you have your window at 1/2 the screen width, you can not log in or start trailing. I think the solution will be to replace the left-side wave icon with a menu button that has all of the actions available. I will merge this once I finish testing and create a new issue for improving the design.

bwhiteman commented:
@bmcdougald you need to treat the logout button the same as the login button, it still goes underneath the start button.
<img width="1431" alt="screen shot 2016-01-19 at 11 13 32 am" src="https://cloud.githubusercontent.com/assets/1516655/12424359/c9305658-be9d-11e5-9b98-4bb8c3942972.png">

bmcdougald commented:
I am not seeing the logout button behavior you are.  Please check your environment, and make sure you rebuilt your toolbar and reinstalled it.  I'm looking at two separate computers here and neither instance of firefox wraps the logout button.  The change involved setting minimum widths on problem columns and a min width on the toolbar as a whole.  The button will scroll off the screen now rather than wrap.  This has to be something in your environment I think.

![screen shot 2016-01-19 at 10 30 56 am](https://cloud.githubusercontent.com/assets/10615545/12424914/e70f0d78-be97-11e5-867a-b2dcd1fcd3a4.png)

As for the comments regarding 1/2 window size...it's unrealistic to think a user would be using our tool and a browser at such a narrow width.  At those widths, toggling the side panel would obscure the entire web page.  So I don't believe the minimum size setting is an issue.  Putting all of the toolbar's functionality into a dropdown menu, not only would require major changes we don't have time for at this moment, but would defeat the concept of a "tool bar".  What you're suggesting may as well be accomplished via a floating button on the page rather than a toolbar, and that's not the direction the team chose when the toolbar work started months ago.

bmcdougald commented:
The behavior you were seeing seems to occur when the username for the logged in user is very long.  This causes the button to resize and that extends it potentially beyond the min-width setting for the div it's contained it.  This causes the wrapping to occur again.  

Mike and I talked this morning about removing the username from the logout button because it caused button resizing which isn't ideal anyway for a toolbar.  I killed two birds with one stone, implemented the change Mike and I discussed and fixed your issue at the same time.
